cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_kernel_default_dense.cpp

```diff
@@ -0,0 +1,138 @@
+        static_cast<backend::interop::decision_forest::interop_model_impl*>(model_pimpl.get());
+
+    daal_input.set(daal::algorithms::classifier::prediction::model, pinterop_model->get_model());
+
+    auto voting_method      = desc.get_voting_method();
+    auto daal_voting_method = voting_method::weighted == voting_method
+                                  ? cls::prediction::weighted
+                                  : cls::prediction::unweighted;
+
+    auto daal_parameter = cls::prediction::Parameter(desc.get_class_count(), daal_voting_method);
+
+    daal::services::Status st;
+
+    daal::data_management::NumericTablePtr daal_prediction_res;
+    daal::data_management::NumericTablePtr daal_prediction_prob_res;
+
+    if (desc.get_infer_results_to_compute() &
+        (std::uint64_t)infer_result_to_compute::compute_class_labels) {
+        daal_prediction_res = interop::allocate_daal_homogen_table<Float>(row_count, 1);
+    }
+
+    if (desc.get_infer_results_to_compute() &
+        (std::uint64_t)infer_result_to_compute::compute_class_probabilities) {
+        daal_prediction_prob_res =
+            interop::allocate_daal_homogen_table<Float>(row_count, desc.get_class_count());
+    }
+
+    const cls::Model* const m = static_cast<cls::Model*>(pinterop_model->get_model().get());
+    interop::call_daal_kernel<Float, cls_default_dense_predict_kernel_t>(
+        ctx,
+        daal::services::internal::hostApp(daal_input),
+        daal_data.get(),
+        m,
+        daal_prediction_res.get(),
+        daal_prediction_prob_res.get(),
+        desc.get_class_count(),
+        daal_voting_method);
+
+    infer_result res;
+
+    if (desc.get_infer_results_to_compute() &
+        (std::uint64_t)infer_result_to_compute::compute_class_labels) {
+        auto table_class_labels =
+            interop::convert_from_daal_homogen_table<Float>(daal_prediction_res);
+        res.set_prediction(table_class_labels);
+    }
+
+    if (desc.get_infer_results_to_compute() &
+        (std::uint64_t)infer_result_to_compute::compute_class_probabilities) {
+        auto table_class_probs =
+            interop::convert_from_daal_homogen_table<Float>(daal_prediction_prob_res);
+        res.set_probabilities(table_class_probs);
+    }
+
+    return res;
+}
+
+template <typename Float>
+static infer_result infer(const context_cpu& ctx,
+                          const descriptor_base& desc,
+                          const infer_input& input) {
+    return call_daal_kernel<Float>(ctx, desc, input.get_model(), input.get_data());
+}
+
+template <typename Float>
+struct infer_kernel_cpu<Float, task::classification, method::default_dense> {
+    infer_result operator()(const context_cpu& ctx,
+                            const descriptor_base& desc,
+                            const infer_input& input) const {
+        return infer<Float>(ctx, desc, input);
+    }
+};
+
+template struct infer_kernel_cpu<float, task::classification, method::default_dense>;
```

Comment:
```suggestion
template struct infer_kernel_cpu<float, task::classification, method::dense>;
```
After discussion yesterday

Reply:
applied